.github/workflows/example-repository-dispatch.yml

```
@@ -0,0 +1,190 @@
+    runs-on: ubuntu-latest
+    if: needs.build.outputs.should_dispatch == 'true'
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check for infrastructure changes
+        id: check_changes
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2046:warning:1:25: Quote this to prevent word splitting

(shellcheck)